make/mt-config/mt-config.mk

```diff
@@ -24,6 +24,7 @@ ifneq (,$(findstring tls_bsp,$(CONFIG)))
 	include make/mt-config/mt-crypto.mk
 endif
 
+include make/mt-config/tests/mt-gtest.mk # For IOTC_GTEST_SOURCES
```

Comment:
this include already appears in [Makefile](https://github.com/GoogleCloudPlatform/iot-device-sdk-embedded-c/blob/master/Makefile#L49). And these includes differ from c or c++ includes since these don't have include guards. Up until now there was an implicit intention to - if possible - only include a makefile once.
We should come up with clear rules how to include .mk file but till then I'd just move the mt-gtest.mk include in Makefile before the mt-config.mk and avoid double .mk file include.
Thoughts?

Reply:
Good point, I did not think about that.

Looks like we have a circular dependency 
- `mt-config.mk` needs `mt-gtest.mk`, for `IOTC_GTEST_SOURCES`
- `mt-gtest.mk` needs `mt-config.mk`, for `LIBIOTC_SRC` and other variables

So just re-ordering the includes in `Makefile` won't help. Either we need to restructure more deeply or have one file will need to include the other (like now). By default I would keep it as it is, since it works, but would gladly reconsider if need be